pytest_picked/modes.py

```diff
@@ -4,26 +4,78 @@
 
 
 class Mode(ABC):
-    def __init__(self, test_file_convention):
-        self.test_file_convention = test_file_convention
+    def __init__(self, test_conventions, only_modified_tests=False):
+        self.test_conventions = test_conventions
+        self.only_modified_tests = only_modified_tests
+
+    def only_tests(self, raw_output, re_list):
+        test_file_regex = r"(\/|^)" + r"|".join(re_list)
+        test_regex = ""
+        class_regex = ""
+        for item in self.test_conventions["function"]:
+            test_regex = (
+                test_regex + r"(?<=def\s).*{0}.*.[\w](?=\(|/:)".format(item) + r"|"
+            )
+        else:
+            test_regex = test_regex[:-1]
+
+        for item in self.test_conventions["class"]:
+            class_regex = (
+                class_regex + r"(?<=class\s).*{0}.*.[\w](?=\(|\:)".format(item) + r"|"
+            )
+        else:
+            class_regex = class_regex[:-1]
+        tests = []
+        last_file_name = None
+        file_dict = {}
```

Comment:
The name `file_dict` makes me think about files only. But this is about storing all the possible tests, right? So I'd suggest renaming this dict to something that is closer to its purpose. Like `tests`, maybe.